### PR TITLE
Create Top Speaker Model #21

### PR DESCRIPTION
**Pull request by MadhushaPrasad**

### CODE REVIEW

File: src/api/interfaces/ITopSpeaker.ts [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+
+interface ITopSpeaker extends Document {
+  speakerName: string;
+  description: string;
+  image_url?: string;
+  socialMedia: ITopSpeakerMedia[];
+  isDelete: boolean;
+}
+
+interface ITopSpeakerMedia {
```
Comment (senuravihanjayadeva) [inline src/api/interfaces/ITopSpeaker.ts:11]: https://github.com/ms-club-sliit/msclubwesite-v2/blob/master/src/data/TopSpeakersData.json

Please Consider above stucture for social Medias

File: src/api/interfaces/ITopSpeaker.ts [hunk 2]
```diff
@@ -0,0 +1,16 @@
+import { Document } from 'mongoose';
+
+interface ITopSpeaker extends Document {
```
Comment (senuravihanjayadeva) [inline src/api/interfaces/ITopSpeaker.ts:3]: Please use the following attribute names when naming 
https://github.com/ms-club-sliit/msclubwesite-v2/blob/master/src/data/TopSpeakersData.json

**Review by senuravihanjayadeva — CHANGES_REQUESTED** [review]

File: src/api/models/TopSpeaker.model.ts [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+  speakerName: { type: String, required: true },
+  description: { type: String, required: false },
+  image_url: { type: String, required: false, default: null },
+  socialMedia: [
+    {
+      name: { type: String, required: true },
+      publicURL: { type: String, required: true },
+    },
+  ],
+  isDelete: { type: Boolean, required: true, default: false },
```
Comment (rusiruavb) [inline src/api/models/TopSpeaker.model.ts:14]: change `isDelete` to `deletedAt` and the data type should be `Date`. 
`deletedAt: { type: Date, required: fasle, default: null }` 
When we get the speaker information just check whether `deletedAt` field is null or not. `null` means its not deleted.

**Review by rusiruavb — CHANGES_REQUESTED** [review]
I have add some comments. Please take a look

**Comment by rusiruavb** [comment]
@MadhushaPrasad please resolve the conflicts

File: src/api/models/TopSpeaker.model.ts (hunk 3 above)
Comment (senuravihanjayadeva) [reply src/api/models/TopSpeaker.model.ts:14]: I though we are using a boolean status. In that case Event modal also should updated.

Comment (rusiruavb) [reply src/api/models/TopSpeaker.model.ts:17]: Normally we use date time to track the deleted time. That's why I suggested to use `Date` for deletedAt.

**Review by rusiruavb — CHANGES_REQUESTED** [review]
change `isDelete` to `deletedAt` and the data type should be `Date`. 
`deletedAt: { type: Date, required: fasle, default: null }` 
When we get the speaker information just check whether `deletedAt` field is null or not. `null` means its not deleted.

**Comment by senuravihanjayadeva** [comment]
please resolve the conflicts @MadhushaPrasad

File: src/api/interfaces/ITopSpeaker.ts (hunk 2 above)
Comment (senuravihanjayadeva) [reply src/api/interfaces/ITopSpeaker.ts:3]: Please Consider attribute names according to the above json File 

    {
      "id": 1,
      "imageUrl": "/assets/speakers/Fiqri-Ismail.jpg",
      "title": "Fiqri Ismail",
      "description": "Architect . Author . Teacher . MVP - Azure",
      "socialMediaURLs": {
        "facebook": "https://www.facebook.com/fiqri.ismail",
        "instagram": "https://www.instagram.com/fiqriismail/",
        "twitter": "https://twitter.com/fiqriismail",
        "linkedIn": "https://www.linkedin.com/pub/fiqri-ismail/29/b75/122",
        "web": "https://askfiqri.com/"
      }

File: src/api/interfaces/ITopSpeaker.ts [hunk 4]
```diff
@@ -1,3 +1,19 @@
...
- */
\ No newline at end of file
+import { Document } from 'mongoose';
+
+interface ITopSpeaker extends Document {
+  speakerName: string;
+  description: string;
+  image_url?: string;
+  socialMedia: ITopSpeakerMedia[];
+  deletedAt?: string;
```
Comment (senuravihanjayadeva) [inline src/api/interfaces/ITopSpeaker.ts:8]: deletedAt datatype should be Date

File: src/api/interfaces/ITopSpeaker.ts [hunk 5]
```diff
@@ -1,3 +1,19 @@
...
- * @todo implement the @interface ITopSpeaker
- */
\ No newline at end of file
+import { Document } from 'mongoose';
+
+interface ITopSpeaker extends Document {
+  speakerName: string;
+  description: string;
+  image_url?: string;
+  socialMedia: ITopSpeakerMedia[];
```
Comment (senuravihanjayadeva) [inline src/api/interfaces/ITopSpeaker.ts:7]: ITopSpeakerMedia not ITopSpeakerMedia[]

Comment (senuravihanjayadeva) [reply src/api/interfaces/ITopSpeaker.ts:7]: socialMediaURLs instead of socialMedia

File: src/api/models/TopSpeaker.model.ts [hunk 6]
```diff
@@ -1,3 +1,25 @@
-/**
- * @todo implement the @model TopSpeaker 
- */
\ No newline at end of file
+import mongoose, { Schema } from 'mongoose';
+import { ITopSpeaker } from '../interfaces';
+
+const TopSpeakerSchema = new Schema<ITopSpeaker>({
```
Comment (senuravihanjayadeva) [inline src/api/models/TopSpeaker.model.ts:4]: Please use the above schema

**Review by senuravihanjayadeva — CHANGES_REQUESTED** [review]

File: src/api/interfaces/ITopSpeaker.ts (hunk 4 above)
Comment (senuravihanjayadeva) [reply src/api/interfaces/ITopSpeaker.ts:8]: https://github.com/ms-club-sliit/msclubwesite-v2/blob/master/src/data/TopSpeakersData.json

consider the following schema when naming attribute

**Review by rusiruavb — CHANGES_REQUESTED** [review]
Make the changes that @senuravihanjayadeva have requested.

**Pull request closed without merging**